plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -59,6 +61,9 @@ public IcebergSplitSource(CloseableIterable<CombinedScanTask> combinedScanIterab
         Iterator<FileScanTask> iterator = limit(fileScanIterator, maxSize);
         while (iterator.hasNext()) {
             FileScanTask task = iterator.next();
+            if (!task.deletes().isEmpty()) {
+                throw new TrinoException(NOT_SUPPORTED, "Iceberg tables with v2 deletes deltas are not supported");
```

Comment:
Nit: I think "v2 deletes deltas" says the same thing several times. Instead, I suggest "Iceberg tables with delete files are not supported."

Reply:
Fair. I wanted to make it clear that we do support v1 copy on write deletes, but I don't have much of a preference.